app/src/main/java/org/openobservatory/ooniprobe/utils/NotificationsRouter.java

```diff
@@ -0,0 +1,70 @@
+package org.openobservatory.ooniprobe.utils;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.activity.BrowserActivity;
+import org.openobservatory.ooniprobe.activity.MainActivity;
+import org.openobservatory.ooniprobe.activity.NotificationDialog;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Map;
+
+public class NotificationsRouter extends FirebaseMessagingService {
+
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+        // TODO(developer): Handle FCM messages here.
+        // Not getting messages here? See why this may be: https://goo.gl/39bRNJ
+        Log.d(TAG, "Message from: " + remoteMessage.getFrom());
+        Log.d(TAG, "Message data payload: " + remoteMessage.getData());
+        Map<String, String> params = remoteMessage.getData();
+
+        // Check if message contains a data payload.
+        if (remoteMessage.getData().size() > 0) {
+            try {
+                //TODO we can use click_action instead of type
```

Comment:
to open the notification from the tray yes, when the app is in foreground we use type